src/agents/workflows/subgraphs/step4_features/solve/graph.py

```
@@ -44,24 +44,53 @@ def _ensure_solve_result(state: AgentState) -> SolveResult:
+) -> tuple[bool, List[str], List[str], str | None, str | None]:
+    """Fallback local execution for generated Python code."""
+
+    temp_file = None
+    try:
+        with tempfile.NamedTemporaryFile(
+            mode="w", suffix=".py", delete=False, encoding="utf-8"
+        ) as handle:
+            handle.write(code)
+            temp_file = handle.name
+
+        result = subprocess.run(
```

Ruff (0.15.0)

[error] `subprocess` call: check for execution of untrusted input

(S603)